dbt/adapters/databricks/python_models/python_submissions.py

```diff
@@ -138,189 +224,318 @@ def _submit_job(self, path: str, cluster_spec: dict) -> str:
             else:
                 libraries.append({"pypi": {"package": package}})
 
-        for lib in additional_libs:
-            libraries.append(lib)
-
-        job_spec.update({"libraries": libraries})
-        run_name = f"{self.database}-{self.schema}-{self.identifier}-{uuid.uuid4()}"
+        for library in additional_libraries:
+            libraries.append(library)
+
+        return {"libraries": libraries}
+
+
+class PythonJobConfigCompiler:
+    """Compiles a Python model into a job configuration for Databricks."""
+
+    def __init__(
+        self,
+        api_client: DatabricksApiClient,
+        permission_builder: PythonPermissionBuilder,
+        run_name: str,
+        cluster_spec: Dict[str, Any],
+        additional_job_settings: Dict[str, Any],
+    ) -> None:
+        self.api_client = api_client
+        self.permission_builder = permission_builder
+        self.run_name = run_name
+        self.cluster_spec = cluster_spec
+        self.additional_job_settings = additional_job_settings
+
+    @staticmethod
+    def create(
+        api_client: DatabricksApiClient,
+        parsed_model: ParsedPythonModel,
+        cluster_spec: Dict[str, Any],
+    ) -> "PythonJobConfigCompiler":
+        permission_builder = PythonPermissionBuilder.create(api_client, parsed_model)
+        packages = parsed_model.config.packages
+        index_url = parsed_model.config.index_url
+        additional_libraries = parsed_model.config.additional_libs
+        library_config = PythonLibraryConfigurer.get_library_config(
+            packages, index_url, additional_libraries
+        )
+        cluster_spec.update(library_config)
```

Comment:
input argument `cluster_spec` should be immutable?

Reply:
yes, that is definitely better.  When I first looked, I didn't see a single line dict merge that didn't mutate, but I just found {**dict1, **dict2}, so I will switch to that.